Feature adamax optimizer

WASAB95 opened the pull request:
AdaMax implemented as an option for the GD optimizer

alperyeg commented:
Hi @WASAB95 thanks for the PR. You uploaded more files then needed for the `adamax` optimizer. Can you reopen a PR only with the essential files, i.e. only the `optimizer.py` and eventually the `init` file.

alperyeg reviewed (changes requested):
Documentation for the adamax optimizer is missing. Unrelated files in the PR.

WASAB95 commented:
The signal generator function was mistakenly added, so I removed it.
the other changes made are necessary to solve the compatibility issue with python 3.10 
otherwise, I added a few essential print commands which were missing in the output of some optimizers
